pic32/libraries/SPI/SPI.h

```diff
@@ -180,10 +183,10 @@ class SPIClass {
 
         inTransactionFlag = 1;
 #endif
-        pspi->sxCon.clr = (1 << _SPICON_ON);
+        // This causes a nasty spike
+        //pspi->sxCon.clr = (1 << _SPICON_ON);
```

Comment:
p.s. I'll be available for a few more hours today if you need anything
right away. I'm done with my pick-n-place machine and reflowing boards for
the evening...

On Fri, Feb 19, 2016 at 1:31 AM, Andrew Kroll <xxxajk@gmail.com> wrote:

> Yes, it should be OK to do that.
> The SPI is idle, not transmitting any data. I've not encountered any
> errors.
> All that the old code did was turn off master mode-- which made it slave
> mode... which you will never be in when you are using transactions. If you
> watch on a 'scope, you'll see the bad side-effect of doing this too... You
> get a phantom SCLK that is a very short pulse, which is:
> 1: Not needed
> 2: could be a source of EMI/RFI
> 3: could confuse some SPI devices.
>
> Would you like me to test different SPI rates and see if it simply works?
> It should be OK to do this since it only happens when there is no data
> transferring, and nothing in the datasheet says anything special is needed.
>
>
>
>
> On Fri, Feb 19, 2016 at 12:11 AM, Brian Schmalz <notifications@github.com>
> wrote:
>
>> In pic32/libraries/SPI/SPI.h
>> <https://github.com/chipKIT32/chipKIT-core/pull/171#discussion_r53423466>
>> :
>>
>> > @@ -180,10 +183,10 @@ class SPIClass {
>> >
>> >          inTransactionFlag = 1;
>> >  #endif
>> > -        pspi->sxCon.clr = (1 << _SPICON_ON);
>> > +        // This causes a nasty spike
>> > +        //pspi->sxCon.clr = (1 << _SPICON_ON);
>>
>> I really like all the rest of these changes. But this one I have a
>> question about: here we are going to be changing the baud rate register
>> (BRG) and the SPI Control register (CON), and it looks like the previous
>> code disabled the SPI port (_SPICON_ON cleared), made the changes, then
>> enabled the SPI port (_SPICON_ON set).
>>
>> So is it OK to change it so that you never clear the _SPICON_ON bit, but
>> just jam new values into BRG and CON without turning the SPI port off and
>> back on again? I couldn't find anything in the PIC32 reference guide that
>> says this is a bad thing, but every one of their pseudo-code sections
>> disables the port, makes changes to these registers, then re-enables the
>> port.
>>
>> Maybe a compromise would be to change the code so that the disable and
>> re-enable of the SPI port is ONLY done if the new BRG or CON values are
>> different from the existing values. Then, I would assume, it would happen
>> once at the first transaction, and never after that unless you changed
>> something.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/chipKIT32/chipKIT-core/pull/171/files#r53423466>.
>>
>
>
>
> --
> Visit my github for awesome Arduino code @ https://github.com/xxxajk
>
>


-- 
Visit my github for awesome Arduino code @ https://github.com/xxxajk